src/modules/currency-converter/services/currency.converter.compute.service.ts

```diff
@@ -0,0 +1,150 @@
+import { TokenComputeService } from 'src/modules/tokens/services/token.compute.service';
+import BigNumber from 'bignumber.js';
+import { cryptoRatesIdentifiers, mxConfig, tokenProviderUSD } from 'src/config';
+import { TokenService } from 'src/modules/tokens/services/token.service';
+
+@Injectable()
+export class CurrencyConverterComputeService {
+    constructor(
+        private readonly apiService: ApiService,
+        private readonly apiConfig: ApiConfigService,
+        private readonly tokenCompute: TokenComputeService,
+        private readonly tokenService: TokenService,
+    ) {}
+
+    @ErrorLoggerAsync()
+    @GetOrSetCache({
+        baseKey: 'currencyConverter',
+        remoteTtl: Constants.oneHour(),
+        localTtl: Constants.oneMinute() * 30,
+    })
+    async allCurrencyRates(): Promise<CurrencyRateModel[]> {
+        const fiatRates = await this.fetchCurrencyRates();
+        const cryptoRates = await this.cryptoRates();
+
+        return [...fiatRates, ...cryptoRates];
+    }
+
+    async cryptoRates(): Promise<CurrencyRateModel[]> {
+        const tokenPrices = await this.tokenCompute.getAllTokensPriceDerivedUSD(
+            cryptoRatesIdentifiers,
+        );
+        const tokens = await this.tokenService.getAllTokensMetadata(
+            cryptoRatesIdentifiers,
+        );
+
+        const tokenRates: CurrencyRateModel[] = cryptoRatesIdentifiers.map(
+            (identifier, index) => {
+                const token = tokens.find((t) => t.identifier === identifier);
+                const rate = tokenPrices[index];
+                const [symbol, name] =
+                    identifier === tokenProviderUSD
+                        ? [mxConfig.EGLDIdentifier, mxConfig.EGLDIdentifier]
+                        : [identifier.split('-')[0], token.name];
+                return {
+                    symbol,
+                    rate: new BigNumber(rate).toNumber(),
+                    category: CurrencyRateType.CRYPTO,
+                    name,
+                };
+            },
+        );
+
+        return tokenRates;
+    }
+
+    async currencyRates(symbols: string[]): Promise<CurrencyRateModel[]> {
+        const allCurrencyRates = await this.allCurrencyRates();
+
+        return allCurrencyRates.filter((rate) => {
+            if (
+                symbols.includes(mxConfig.EGLDIdentifier) &&
+                rate.symbol === tokenProviderUSD
+            ) {
+                return true;
+            }
+            return symbols.includes(rate.symbol);
+        });
+    }
+
+    async fetchCurrencyRates(symbols?: string[]): Promise<CurrencyRateModel[]> {
+        const apiEndpoint = this.apiConfig.getOpenExchangeRateUrl();
+        const params: any = {
+            app_id: this.apiConfig.getOpenExchangeRateAppID(),
+            base: 'USD',
+        };
+
+        if (symbols && symbols.length > 0) {
+            params.symbols = symbols.join(',');
+        }
+
+        try {
```

Comment:
I would wrap this whole method into the try catch. The backend should operate normally even without an `APP_ID`.